### PR TITLE
[#3976] improvement(core): Improve the exception behavior in entity store

### PR DESCRIPTION
**Pull request by xloya**
### What changes were proposed in this pull request?

Improve the throwing behavior of some SQL exceptions in EntityStore.

### Why are the changes needed?

Fix: #3976 

### How was this patch tested?

Through the existing tests.

### CODE REVIEW

**Comment by jerryshao** [comment]
@yuqi1129 please help to review, thanks.

File: core/src/main/java/com/datastrato/gravitino/storage/relational/converters/SQLExceptionConverter.java [hunk 1]
```diff
@@ -16,9 +16,7 @@ public interface SQLExceptionConverter {
    * @param sqlException The sql exception to map
    * @param type The type of the entity
    * @param name The name of the entity
-   * @return A best attempt at a corresponding jdbc connector exception or generic with the
-   *     SQLException as the cause
    */
-  GravitinoRuntimeException toGravitinoException(
-      SQLException sqlException, Entity.EntityType type, String name);
+  void toGravitinoException(SQLException sqlException, Entity.EntityType type, String name)
```
Comment (yuqi1129) [inline core/src/main/java/com/datastrato/gravitino/storage/relational/converters/SQLExceptionConverter.java:20]: Why do you change the return type here and directly throw an exception?

Comment (yuqi1129) [reply core/src/main/java/com/datastrato/gravitino/storage/relational/converters/SQLExceptionConverter.java:20]: I see.

**Comment by yuqi1129** [comment]
@jerryshao 
I'm okay with the changes. Do you need to review them?

**Review by jerryshao — APPROVED** [review]
LGTM, thanks @xloya for your work.

**Pull request merged**